ddl/ddl_api.go

```diff
@@ -3561,6 +3561,9 @@ func checkColumnWithIndexConstraint(tbInfo *model.TableInfo, originalCol, newCol
 }
 
 func checkAutoRandom(tableInfo *model.TableInfo, originCol *table.Column, specNewColumn *ast.ColumnDef) (uint64, error) {
+	if tableInfo.GetPkName().L != originCol.Name.L {
```

Comment:
Why not check HasPriKeyFlag?

Reply:
`GetPkName` checks `HasPriKeyFlag` internally.

Reply:
tableInfo.GetPkName() return one column only? What if multi-column primary key?

Reply:
Good catch!